aioresponses/compat.py

```diff
@@ -49,11 +55,97 @@ def normalize_url(url: 'Union[URL, str]') -> 'URL':
     return url.with_query(urlencode(sorted(parse_qsl(url.query_string))))
 
 
+def _build_raw_headers(headers: Dict) -> Tuple:
+    """
+    Convert a dict of headers to a tuple of tuples
+
+    Mimics the format of ClientResponse.
+    """
+    raw_headers = []
+    for k, v in headers.items():
+        raw_headers.append((k.encode('utf8'), v.encode('utf8')))
+    return tuple(raw_headers)
+
+
+def build_response(
+        url: 'Union[URL, str]',
+        method: str = hdrs.METH_GET,
+        status: int = 200,
+        body: str = '',
+        content_type: str = 'application/json',
+        payload: Dict = None,
+        headers: Dict = None,
+        response_class: 'ClientResponse' = None,
+        reason: Optional[str] = None) -> ClientResponse:
+    if response_class is None:
+        response_class = ClientResponse
+    if payload is not None:
+        body = json.dumps(payload)
+    if not isinstance(body, bytes):
+        body = str.encode(body)
+    kwargs = {}
+    if AIOHTTP_VERSION >= StrictVersion('3.1.0'):
+        loop = Mock()
+        loop.get_debug = Mock()
+        loop.get_debug.return_value = True
+        kwargs['request_info'] = Mock()
+        kwargs['writer'] = Mock()
+        kwargs['continue100'] = None
+        kwargs['timer'] = TimerNoop()
+        if AIOHTTP_VERSION < StrictVersion('3.3.0'):
+            kwargs['auto_decompress'] = True
+        kwargs['traces'] = []
+        kwargs['loop'] = loop
+        kwargs['session'] = None
+    _headers = CIMultiDict({hdrs.CONTENT_TYPE: content_type})
+    if headers:
+        _headers.update(headers)
+    raw_headers = _build_raw_headers(_headers)
+    resp = response_class(method, url, **kwargs)
+    if AIOHTTP_VERSION >= StrictVersion('3.3.0'):
+        # Reified attributes
+        resp._headers = _headers
+        resp._raw_headers = raw_headers
+    else:
+        resp.headers = _headers
+        resp.raw_headers = raw_headers
+    resp.status = status
+    resp.reason = reason
+    resp.content = stream_reader_factory()
+    resp.content.feed_data(body)
+    resp.content.feed_eof()
+    return resp
+
+
+class CallbackResult:
```

Comment:
Could you please move 
`_build_raw_headers ` and `build_response` methods back to RequestMatch class ?

Also `CallbackResult` should be in `core` module.

Reply:
Done.